hardware/arduino/sam/cores/arduino/wiring_digital.c

```diff
@@ -49,15 +64,18 @@ extern void pinMode( uint32_t ulPin, uint32_t ulMode )
             	PIO_INPUT,
             	g_APinDescription[ulPin].ulPin,
             	PIO_PULLUP ) ;
+            g_pinStatus[ulPin] = (g_pinStatus[ulPin] & 0xF0) | PIN_STATUS_DIGITAL_INPUT_PULLUP;
         break ;
 
         case OUTPUT:
             PIO_Configure(
             	g_APinDescription[ulPin].pPort,
-            	PIO_OUTPUT_1,
+              g_pinStatus[ulPin] = ((g_pinStatus[ulPin] & 0xF0) >> 4 ? PIO_OUTPUT_1 : PIO_OUTPUT_0),
```

Comment:
There is no need to set g_pinStatus if you are going to do it again on the next line.  There is no need to do `& 0xF0` because those bits you are masking off would get shifted anyway.